pandas/io/excel/_calamine.py

```diff
@@ -99,26 +98,31 @@ def get_sheet_by_index(self, index: int) -> CalamineSheet:
 
     def get_sheet_data(
         self, sheet: CalamineSheet, file_rows_needed: int | None = None
-    ) -> list[list[Scalar | NaTType | time]]:
-        def _convert_cell(value: _CellValue) -> Scalar | NaTType | time:
+    ) -> list[list[Scalar]]:
+        def _convert_cell(value: _CellValueT) -> Scalar:
+            # Avoid explicit conversion to pd.Timestamp and pd.Timedelta
             if isinstance(value, float):
                 val = int(value)
                 if val == value:
                     return val
                 else:
                     return value
             elif isinstance(value, date):
-                return pd.Timestamp(value)
+                return value
             elif isinstance(value, timedelta):
-                return pd.Timedelta(value)
-            elif isinstance(value, time):
                 return value
+            elif isinstance(value, time):
```

Comment:
I think these can just be removed. We only need `elif instance(value, ...)` if we are going to do something besides just returning `value`.